src/librustdoc/clean/auto_trait.rs

```diff
@@ -99,29 +131,126 @@ impl<'a, 'tcx, 'rcx> AutoTraitFinder<'a, 'tcx, 'rcx> {
         let tcx = self.cx.tcx;
         let generics = self.cx.tcx.generics_of(def_id);
 
+        let ty = self.cx.tcx.type_of(def_id);
+        let mut traits = Vec::new();
+        if self.cx.crate_name != Some("core".to_string()) &&
+           self.cx.access_levels.borrow().is_doc_reachable(def_id) {
+            if let ty::TyAdt(_adt, _) = ty.sty {
+                let real_name = name.clone().map(|name| Ident::from_str(&name));
+                let param_env = self.cx.tcx.param_env(def_id);
+                for &trait_def_id in self.cx.all_traits.iter() {
+                    if !self.cx.access_levels.borrow().is_doc_reachable(trait_def_id) ||
```

Comment:
I think it would make sense to prefilter `all_traits` during creation by their reachability instead of doing this over and over again for every single type in a crate

Reply:
This is something I discussed a bit earlier with @QuietMisdreavus so I'll do it as well.

Reply:
Actually I can't, it depends on the context so we have to make this check in here (`access_levels` is updated through the code).